File: valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parantheses = {'(': ')', '{': '}', '[': ']'}
        stack = []
        for char in s:
            if char in parantheses.keys():
                stack.append(char)
            else:
                if len(stack) != 0 and parantheses[stack[-1]] == char:
                    stack.pop()
                else:
                    return False
        if len(stack) == 0:
            return True
        else:
            return False

File: valid-parentheses/valid-parentheses.go
// convertion from gpt
func isValid(s string) bool {
    parentheses := map[rune]rune{'(': ')', '{': '}', '[': ']'}
    stack := make([]rune, 0)
    for _, char := range s {
        if _, ok := parentheses[char]; ok {
            stack = append(stack, char)
        } else if len(stack) == 0 || parentheses[stack[len(stack)-1]] != char {
            return false
        } else {
            stack = stack[:len(stack)-1]
        }
    }
    return len(stack) == 0
}